components/renku_data_services/notifications/api.spec.yaml

```diff
@@ -0,0 +1,204 @@
+          application/json:
+            schema:
+              $ref: "#/components/schemas/AlertPost"
+      responses:
+        "201":
+          description: Alert successfully created
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Alert"
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - alerts
+    get:
+      summary: Retrieve all active alerts for the authenticated user.
+      parameters:
+        - in: query
+          description: query parameters
+          name: params
+          style: form
+          explode: true
+          schema:
+            $ref: "#/components/schemas/AlertsGetQuery"
+      responses:
+        "200":
+          description: A list of active alerts for the authenticated user.
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/AlertList"
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - alerts
+  /alerts/{alert_id}:
+    patch:
+      summary: Resolve an alert. Requires admin permissions.
+      description: Mark an alert as resolved. The alert remains in the database but is no longer active.
+      parameters:
+        - in: path
+          name: alert_id
+          required: true
+          schema:
+            $ref: "#/components/schemas/Ulid"
+          description: The ID of the alert to resolve.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/AlertPatch"
+      responses:
+        "200":
+          description: Alert successfully updated
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Alert"
+        "404":
+          description: Alert not found
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - alerts
+components:
+  schemas:
+    AlertsGetQuery:
+      description: Query parameters for getting alerts
+      type: object
+      additionalProperties: false
+      properties:
+        session_name:
+          $ref: "#/components/schemas/SessionName"
+    AlertPatch:
+      description: Data to update an alert.
+      type: object
+      additionalProperties: false
+      properties:
+        resolved:
+          type: boolean
+          description: Set to true to mark the alert as resolved. The resolved_at timestamp will be set automatically.
+    Ulid:
+      description: ULID identifier
+      type: string
+      minLength: 26
+      maxLength: 26
+      pattern: "^[0-7][0-9A-HJKMNP-TV-Z]{25}$" # This is case-insensitive
+    Alert:
+      description: An alert sent or displayed to a user.
+      type: object
+      properties:
```

Comment:
Note, can be updated in a later PR: I think that maybe we could have a `status: firing/resolved` field. This way, patching an alert would use clean REST semantics (that is patching an existing field with an allowed value).